yaptide/routes/common_sim_routes.py

```diff
@@ -98,6 +98,36 @@ def post():
         return yaptide_response(message="Task updated", code=202)
 
 
+def get_single_estimator(sim_id: int, estimator_name: str):
+    """Retrieve a single estimator by simulation ID and estimator name"""
+    estimator = fetch_estimator_by_sim_id_and_est_name(sim_id=sim_id, est_name=estimator_name)
+    if not estimator:
+        return yaptide_response(message="Estimator not found", code=404)
+
+    pages = fetch_pages_by_estimator_id(est_id=estimator.id)
+    estimator_dict = {"metadata": estimator.data, "name": estimator.name, "pages": [page.data for page in pages]}
+    return yaptide_response(message=f"Estimator '{estimator_name}' for simulation: {sim_id}",
+                            code=200,
+                            content=estimator_dict)
+
+
+def get_all_estimators(sim_id: int):
+    """Retrieve all estimators for a given simulation ID"""
+    estimators = fetch_estimators_by_sim_id(sim_id=sim_id)
+    if len(estimators) == 0:
+        return yaptide_response(message="Results are unavailable", code=404)
+
+    logging.debug("Returning results from database")
+    result_estimators = []
+    for estimator in estimators:
+        pages = fetch_pages_by_estimator_id(est_id=estimator.id)
+        estimator_dict = {"metadata": estimator.data, "name": estimator.name, "pages": [page.data for page in pages]}
+        result_estimators.append(estimator_dict)
```

Comment:
This will be pretty slow if we would have thousdands of estimators, as this code would do thousands of queries to the DB.

I'd strongly recomment method like `fetch_estimators_and_pages_by_simulation_id` which would make **a single query to the DB** and going over the results of that single query would return something that you have here in the `estimator_dict`.